internal/users/manager.go

```diff
@@ -33,15 +37,55 @@ var DefaultConfig = Config{
 	GIDMax: 1999999999,
 }
 
+type temporaryUser struct {
+	name    string
+	preAuth bool
+}
+
 // Manager is the manager for any user related operation.
 type Manager struct {
-	cache  *cache.Cache
-	config Config
+	cache                 *cache.Cache
+	config                Config
+	temporaryEntriesMu    sync.Mutex
```

Comment:
Would it make sense to use a `sync.RWMutex` instead?

Reply:
Why?

Reply:
> Why?

See https://github.com/ubuntu/authd/pull/663#discussion_r1890087354

> Would it make sense to use a sync.RWMutex instead?

Done